src/Extensions/ElementalAreasExtension.php

```diff
@@ -229,13 +229,23 @@ public function onBeforeWrite()
 
         $ownerClassName = get_class($this->owner);
 
-        // Update the OwnerClassName on EA if the class has changed
         foreach ($elementalAreaRelations as $eaRelation) {
+            // Update the OwnerClassName on EA if the class has changed
+            /** @var ElementalArea $ea */
             $ea = $this->owner->$eaRelation();
             if ($ea->OwnerClassName !== $ownerClassName) {
                 $ea->OwnerClassName = $ownerClassName;
                 $ea->write();
             }
+
+            // Owner page will have set its own HasBrokenLink / HasBrokenFile in SiteTreeLinkTracking / FileLinkTracking
+            // on any $page->Content BEFORE we get to this point
+            if (in_array(1, $ea->Elements()->column('HasBrokenLink'))) {
```

Comment:
May I suggest using `if (array_filter($ea->Elements()->column('HasBrokenLink'))` to support any truthy value rather than check for integer value `1` in the list?

Reply:
I've updated this to check for truthy values in arrays